amivapi/ldap.py

```diff
@@ -28,9 +28,10 @@
 
 def init_app(app):
     """Attach an ldap connection to the app."""
-    user = app.config['LDAP_USER']
-    password = app.config['LDAP_PASS']
-    app.config['ldap_connector'] = AuthenticatedLdap(user, password)
+    user = app.config['LDAP_USERNAME']
+    password = app.config['LDAP_PASSWORD']
+    if user and password:
```

Comment:
What does this `if` actually do?
The only case I see where this becomes relevant is in case `LDAP_USERNAME` and `LDAP_PASSWORD` are actually set, but evaluate to some statement that is false (empty strings, `False`, ...). This would be a weird case, but it would be skipped quite intransparantly to the user (because nothing would happen, but LDAP would also not work).
Is there any other case I do not see which is a reason for this `if` statement?

Reply:
The behavior is intended, if the strings are empty (no username and/or password), the eth ldap won't work.

The fields are never not set, since `settings.py` contains defaults.

I could change the defaults to `None`, but not sure if that would help.

I think I will add some logging to inform the user whether ldap was enabled.

Reply:
What would be the problem if ldap simply fails because of empty strings? I feel like that would be more transparent behaviour

Reply:
Ok, I have changed the defaults to None and use whatever value specified if both username and password are not None